drivers/net/native_sim_sockets.c

```diff
@@ -0,0 +1,533 @@
+/**
+ * Copyright (c) 2023 Marcin Niestroj
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/net/ethernet.h>
+#include <zephyr/net/net_ip.h>
+#include <zephyr/net/offloaded_netdev.h>
+#include <zephyr/net/socket_offload.h>
+#include <zephyr/sys/fdtable.h>
+
+#include "sockets_internal.h"
+#include "native_sim.h"
+#include "native_sim_errno.h"
+
+/* Increment by 1 to make sure we do not store the value of 0, which has
+ * a special meaning in the fdtable subsys.
+ */
+#define FD_TO_OBJ(fd) ((void *)(intptr_t)((fd) + 1))
+#define OBJ_TO_FD(obj) (((int)(intptr_t)(obj)) - 1)
+
+static int socket_family_to_nsim(int family, int *family_nsim)
+{
+	switch (family) {
+	case AF_UNSPEC:
+		*family_nsim = NSIM_AF_UNSPEC;
```

Comment:
I am getting all the time confused with the native word in various places. It does not give much context, at least for me. It would be great if it is obvious where the code is to be used (zephyr vs host side).

Reply:
I have changed all those "middleground" macros and conversion functions to use `NSOS_MID_` and `nsos_mid_`. I hope that improves readability.